src/harness/reference_models/interference/interference.py

```diff
@@ -0,0 +1,596 @@
+from reference_models.examples import entities, example_fss_interference
+from collections import namedtuple
+from enum import Enum
+from shapely.geometry import Point as SPoint
+from shapely.geometry import Polygon as SPolygon
+from shapely.geometry import MultiPolygon as MPolygon
+from shapely.geometry import shape, Point, LineString
+
+# Initialize terrain driver
+# terrainDriver = terrain.TerrainDriver()
+terrainDriver = wf_itm.terrainDriver
+
+# Set constant parameters based on requirements in the WINNF-TS-0112
+# [R2-SGN-16]
+GWPZ_NBRHD_DIST = 40  # neighborhood distance from a CBSD to a given protection
+# point (in km) in GWPZ protection area
+PPA_NBRHD_DIST = 40  # neighborhood distance from a CBSD to a given protection
+# point (in km) in PPA protection area
+
+FSS_CO_CHANNEL_NBRHD_DIST = 150  # neighborhood distance from a CBSD to FSS for
+# co-channel protection
+
+FSS_BLOCKING_NBRHD_DIST = 40  # neighborhood distance from a CBSD to FSS
+# blocking protection
+
+ESC_NBRHD_DIST_A = 40  # neighborhood distance from a ESC to category A CBSD
+
+ESC_NBRHD_DIST_B = 80  # neighborhood distance from a ESC to category B CBSD
+
+NUM_OF_PROCESSES = 6
+
+# Frequency used in propagation model (in MHz) [R2-SGN-04]
+FREQ_PROP_MODEL = 3625.0
+
+# CBRS Band Frequency Range (Hz)
+CBRS_LOW_FREQ = 3550.e6
+CBRS_HIGH_FREQ = 3700.e6
+
+# FSS Passband low frequency range  (Hz)
+FSS_LOW_FREQ = 3600.e6
+
+# FSS Passband for TT&C (Hz)
+FSS_TTC_LOW_FREQ = 3700.e6
+FSS_TTC_HIGH_FREQ = 4200.e6
+
+# ESC IAP for Out-of-Band Categoroy A CBSDs in Frequency Range (Hz)
+ESC_CAT_A_LOW_FREQ = 3550.e6
+ESC_CAT_A_HIGH_FREQ = 3660.e6
+
+# ESC IAP for Out-of-Band Categoroy B CBSDs in Frequency Range (Hz)
+ESC_CAT_B_LOW_FREQ = 3550.e6
+ESC_CAT_B_HIGH_FREQ = 3680.e6
+
+# ESC Channel 21 Center Frequency
+ESC_CH21_CF = 36525.e5
+
+# One Mega Hertz
+ONE_MHZ = 1.e6
+
+# Channel bandwidth over which SASs execute the IAP process
+IAPBW = 5.e6
+
+# GWPZ Area Protection reference bandwidth for the IAP process
+GWPZ_RBW = 10.e6
+
+# PPA Area Protection reference bandwidth for the IAP process
+PPA_RBW = 10.e6
+
+# GWPZ and PPA height (m)
+GWPZ_PPA_HEIGHT = 1.5
+
+# Number of SASs - NSAS
+NUM_SAS = 3
+
+# Global grant counter
+grant_counter = 0
+
+# In-band insertion loss
+IN_BAND_INSERTION_LOSS = 0.5
+
+# Define an enumeration class named ProtectionEntityType with members
+# 'GWPZ_AREA', 'PPA_AREA', 'FSS_CO_CHANNEL', 'FSS_BLOCKING', 'ESC_CAT_A'
+# 'ESC_CAT_B'
+
+
+class ProtectionEntityType(Enum):
+  GWPZ_AREA = 1
+  PPA_AREA = 2
+  FSS_CO_CHANNEL = 3
+  FSS_BLOCKING = 4
+  ESC_CAT_A = 5
+  ESC_CAT_B = 6
+
+
+# Define CBSD grant, i.e., a tuple with named fields of 'latitude',
+# 'longitude', 'height_agl', 'indoor_deployment', 'antenna_azimuth',
+# 'antenna_gain', 'antenna_beamwidth', 'cbsd_category', 'grant_index',
+# 'max_eirp', 'low_frequency', 'high_frequency', 'is_managed_grant'
+CbsdGrantInformation = namedtuple('CbsdGrantInformation',
```

Comment:
getAllGrantInformationFromCbsdDataDump()   API creates 2 tuples of type CbsdGrantInformation if 2 grants are associated   with CBSD           CbsdGrantInformation is used to store some additional information needed   for IAP calculation. Information such as is_managed_grant and grant_index are   the additional parameters. These parameters are used to calculate A_SAS_P.             This new tuple is used mainly because for storing the additional   information for each grant and for easier access to all the parameters   related to a grant (like the lat, long and other installation params).